### PR TITLE
WIP: product solutions docs

### PR DESCRIPTION
**Pull request by sqs**
These are some of the "use case docs" that I outlined at https://docs.google.com/document/d/1NoWBGbXKlrpwBWyk7o4hjzTEzVHf2ST4FoJiyneW8DA/edit#heading=h.g18bnyd8l68w.

### CODE REVIEW

**Comment by beyang** [comment]
- Mention knowledge transfer. Code review is not just about ensuring correctness (automated tests are often a better way of doing that), but also about ensuring that the collective knowledge of your team reflects the current state of the codebase. Robust software requires robust dissemination of knowledge among the software team.
- Note there is a diversity of code review tools and processes—and that there's no one-size-fits-all solution. Sourcegraph is not offering Yet Another Code Review tool, but an intelligent plugin that enhances any code review tool.
- Another important scenario for code review is onboarding/mentoring a new/junior teammate. Typically the junior person writes the code and the senior person reviews. High-quality feedback from the senior person is crucial to ensuring the junior person's code meets team standards *and* leveling up the junior person's skills. However, senior people's time is both *valuable* and *limited*. Sourcegraph enables them to give *more* code reviews of *higher quality* in a *shorter amount of time*. Transform code review from a tax on senior developers' time to a true force multiplier for the entire team.

File: doc/dev/product/solutions/code_review.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+Code review is a fundamental process of any development team. It's important for improving quality, sharing knowledge, and avoiding mistakes.
+
+No matter what code review tool you use ([GitHub](https://github.com), [GitLab](https://gitlab.com), [Bitbucket](https://bitbucket.org), [Phabricator](https://phabricator.org), or [something else](TODO!(sqs): link to code review tools list)), the effectiveness of code review depends on human factors:
+
+- Does the reviewer understand the changed code?
+- Is it clear how the change change affects the entire system (including other services and/or repositories)?
+<!-- - Are the right developers reviewing the code? TODO(sqs): This involves saved searches. It can be added to this doc later. -->
+
+This document explains how [Sourcegraph](https://sourcegraph.com) helps with these important parts of your code review process. Sourcegraph is open-source and self-hosted, and it integrates with all the code review tools listed above. You can [start using it](../../../index.md#quickstart) in a few minutes for your organization.
```
Comment (nicksnyder) [inline doc/dev/product/solutions/code_review.md]: Candidate for style guide
- [open-source](https://sourcegraph.sgdev.org/search?q=%22open-source%22+count:1000)
- [open source](https://sourcegraph.sgdev.org/search?q=%22open+source%22+count:1000)

File: doc/dev/product/solutions/code_review.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+### Problem
+
+To review changed code effectively, you need to actually understand the code:
+
+- What calls what?
+- Are the right arguments given?
+- Are errors from function calls checked correctly?
+- Etc.
+
+<!-- TODO(sqs): I don't like how the bullets above turned out. Will revisit (suggestions appreciated). -->
```
Comment (nicksnyder) [inline doc/dev/product/solutions/code_review.md]: Most commonly for me:
- "The code starts to use a function or variable that already exists (i.e. it is not part of the diff). What is that function documented to do? What does it actually do?"
- "I want to jump around the diff to view where functions are used/defined. Without Sourcegraph I use find-in-page text search (which doesn't work if diffs are auto collapsed)."

File: doc/dev/product/solutions/code_review.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+You wouldn't *write* code without this information, so why would you *review* code without it? This information is usually available in editors (using go-to-definition, find-references, and hover tooltips), but none of the code review tools above offer it.
+
+You could check out the branch for the code review locally and open the code in your editor to get this information, but in our experience, that's rarely done. If you're the rare, diligent developer who always does that, then we applaud you. But your teammates likely aren't so diligent, and wouldn't it be great to save a few minutes each time you do a review (and be able to add review comments without switching back and forth)?
```
Comment (nicksnyder) [inline doc/dev/product/solutions/code_review.md]: The tone in this paragraph and previous paragraph feels a bit two preachy. I think it is sufficient to lay out the options and let the reader make the judgement.

- Review code in your browser without go-to-definition, find-references, and hover tooltips.
- Stash any local changes that you have, check out the review branch, and open it in your editor for code intelligence on that repository.
- Use Sourcegraph to get go-to-definition, find-references, and hover tooltips while reviewing code in your browser on your code host.

**Comment by sqs** [comment]
@ryan-blunden 	Just removed the onboarding doc from this PR. You now own this PR. :)

**Pull request closed without merging**